experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
+        :param: exp_slice: a slice to check
+        """
+        error_count = 0
+        error_dict = {}
+
+        options = self.options
+
+        for param in self.slice_keys:
+            try:
+                 assert param in exp_slice.keys()
+            except AssertionError:
+                if param == 'txfreq' and exp_slice['clrfrqflag']:
+                    pass
+                elif param == 'rxfreq' and not exp_slice['rxonly']:
+                    pass
+                elif param == 'clrfrqrange' and not exp_slice['clrfrqflag']:
+                    pass
+                else:
+                    errmsg = "Slice {} is missing Necessary Parameter {}".format(
+                        exp_slice['slice_id'], param)
+                    raise ExperimentException(errmsg)
+                    # set defaults if possible
+            try:
+                assert param is not None
+            except AssertionError:
+                pass  # TODO may want to check certain params are not None
+
+        if error_dict:  # if not empty
+            return error_dict  # cannot check all params because they don't exist
+
+        for param in exp_slice.keys():
+            if param not in self.slice_keys and param not in self.__hidden_slice_keys:
+                error_dict[error_count] = "Slice {} has A Parameter that is not Used: {} = {}". \
+                    format(exp_slice['slice_id'], param, exp_slice[param])
+                error_count += 1
+
+        if len(exp_slice['txantennas']) > options.main_antenna_count:
+            error_dict[
+                error_count] = "Slice {} Has Too Many Main TX Antenna Channels {} Greater than Config {}" \
+                .format(exp_slice['slice_id'], len(exp_slice['txantennas']),
+                        options.main_antenna_count)
+        if len(exp_slice['rx_main_antennas']) > options.main_antenna_count:
+            error_dict[
+                error_count] = "Slice {} Has Too Many Main RX Antenna Channels {} Greater than Config {}" \
+                .format(exp_slice['slice_id'], len(exp_slice['rx_main_antennas']),
+                        options.main_antenna_count)
+            error_count = error_count + 1
+        if len(exp_slice['rx_int_antennas']) > options.interferometer_antenna_count:
+            error_dict[
+                error_count] = "Slice {} Has Too Many RX Interferometer Antenna Channels {} " \
+                               "Greater than Config {}".format(
+                                    exp_slice['slice_id'],
+                                    len(exp_slice['rx_int_antennas']),
+                                    options.interferometer_antenna_count)
+            error_count = error_count + 1
+
+        # Check if the antenna identifier number is greater than the config file's
+        # maximum antennas for all three of tx antennas, rx antennas and rx int antennas
+        # Also check for duplicates
+        for i in range(len(exp_slice['txantennas'])):
```

Comment:
Can just check max(...) and see that it's not greater than options.main_antenna_count